cmd/sync-rover-groups/ldap.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	ldapv3 "github.com/go-ldap/ldap/v3"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+type ldapGroupResolver struct {
+	conn ldapConn
+}
+
+type ldapConn interface {
+	Search(searchRequest *ldapv3.SearchRequest) (*ldapv3.SearchResult, error)
+}
+
+const (
+	baseDN = "dc=redhat,dc=com"
+)
+
+type notFoundErr struct {
+	group string
+}
+
+// IsNotFoundError returns true if the error indicates the provided
+// object is not found.
+func IsNotFoundError(err error) bool {
+	if err == nil {
+		return false
+	}
+	_, ok := err.(*notFoundErr)
+	return ok
+}
+
+func (e *notFoundErr) Error() string {
+	return fmt.Sprintf("failed to find group on the ldap server: %s", e.group)
+}
+
+func (r *ldapGroupResolver) resolve(name string) (*Group, error) {
+	filter := fmt.Sprintf("(&(objectClass=rhatRoverGroup)(cn=%s))", ldapv3.EscapeFilter(name))
+	searchReq := ldapv3.NewSearchRequest(baseDN, ldapv3.ScopeWholeSubtree, 0, 0, 0, false, filter, []string{"uniqueMember"}, []ldapv3.Control{})
+
+	result, err := r.conn.Search(searchReq)
+	if err != nil {
+		return nil, fmt.Errorf("failed to search ldap: %w", err)
+	}
+
+	if len(result.Entries) == 0 {
+		return nil, &notFoundErr{group: name}
+	}
+
+	if l := len(result.Entries); l > 1 {
+		// this should never happen
+		return nil, fmt.Errorf("found %d group with the name: %s", l, name)
+	}
```

Comment:
these both deal with len() so we could do:

```go
switch items := len(); {
case items == 0:
  return something
case items > 1:
  return something else
}
```

Reply:
Updated.